mlflow/models/evaluation/base.py

```diff
@@ -496,33 +685,72 @@ def evaluate(
                              a nested dictionary whose key is the evaluator name.
     :return: An :py:class:`mlflow.models.evaluation.EvaluationDataset` instance containing
              evaluation results.
+
```

Comment:
Nice! I really like the documentation here to spell out the expected outcome and limitations. Hopefully we'll make it shorter and easier to navigate over time.